### PR TITLE
[Analytics] Displays location details for clicked map emojis

### PR DESCRIPTION
**Pull request by Codebmk**
#### Summary of Changes (What does this PR do?)

- Added ability to click a map emoji and view details about that location. The details include the weekly air quality predictions, and custom air quality alert
#### Status of maturity (all need to be checked before merging):

- [x] I've tested this locally
- [x] I consider this code done
- [x] This change ready to hit production in its current state

#### How should this be manually tested?

- Please include the steps to be done inorder to setup and test this PR.

#### What are the relevant tickets?

- [AN-611](https://airqoteam.atlassian.net/browse/AN-611) 
- [AN-615](https://airqoteam.atlassian.net/browse/AN-615)

#### Screenshots (optional)
image

[AN-611]: https://airqoteam.atlassian.net/browse/AN-611?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

[AN-615]: https://airqoteam.atlassian.net/browse/AN-615?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: platform/src/common/components/Map/components/Sidebar.js [hunk 1]
```diff
@@ -229,7 +232,7 @@ const WeekPrediction = ({ siteDetails, currentDay, airQualityReadings, weekDays
 
   return (
     <div className='relative' ref={dropdownRef}>
-      <div className='mb-5 relative'>
+      {/* <div className='mb-5 relative'>
```
Comment (Baalmart) [inline platform/src/common/components/Map/components/Sidebar.js:235]: Do we plan to use this commented code in the near future?

File: platform/src/common/components/Map/components/Sidebar.js [hunk 2]
```diff
@@ -658,22 +741,25 @@ const Sidebar = ({ siteDetails, selectedSites, isAdmin, showSideBar, setShowSide
...
+                  title='Air Quality Alerts'
+                  isCollapsed
+                  children={
+                    <p className='text-xl font-bold leading-7 text-secondary-neutral-dark-950'>
+                      <span className='text-blue-500 capitalize'>
+                        {selectedSite?.place_name ||
+                          selectedSite?.name?.split(',')[0] ||
+                          selectedSite?.search_name ||
+                          selectedSite?.location}
+                        's
```
Comment (Baalmart) [inline platform/src/common/components/Map/components/Sidebar.js:755]: this "'s" looks suspicious, is it supposed to be there?

File: platform/src/common/components/Map/components/MapNodes.js [hunk 3]
```diff
@@ -103,6 +103,30 @@ export const getAQIcon = (pollutant, value) => {
   }
 };
 
+export const getAQIMessage = (pollutant, value) => {
+  if (!markerDetails.hasOwnProperty(pollutant)) {
+    throw new Error(`Invalid pollutant: ${pollutant}`);
+  }
+
+  const aqiCategory = getAQICategory(pollutant, value);
```
Comment (Baalmart) [inline platform/src/common/components/Map/components/MapNodes.js:111]: Hi @Codebmk , this is amazing. We could refine this code further.....

We can use a combination of a lookup object and the optional chaining operator (?.). This approach simplifies the code by removing the repetitive if-else structure and makes it easier to add or modify conditions in the future. Additionally, the optional chaining operator is supported in modern JavaScript environments,but it's not available in older versions of JavaScript so we might want to rethink things as we go about its usage --- it is supported in Node.js versions 14 and above, and in most modern browsers. However, it's not available in older JavaScript environments.

Here's how we can refactor the code:

```
const aqiMessages = {
 'GoodAir': 'Enjoy the day with confidence in the clean air around you.',
 'ModerateAir': 'Today is a great day for an outdoor activity.',
 'UnhealthyForSensitiveGroups': 'Reduce the intensity of your outdoor activities.',
 'Unhealthy': 'Avoid activities that make you breathe more rapidly. Today is the perfect day to spend indoors reading.',
 'VeryUnhealthy': 'Reduce the intensity of your outdoor activities. Try to stay indoors until the air quality improves.',
 'Hazardous': 'If you have to spend a lot of time outside, disposable masks like the N95 are helpful.',
};

const aqiCategory = getAQICategory(pollutant, value);

return aqiMessages[aqiCategory?.icon] || '';
```

**Review by Baalmart — COMMENTED** [review]
Thanks @Codebmk , this is great work...thanks a lot! 
I have just left a few comments for your consideration and feedback.

File: platform/src/common/components/Map/components/Sidebar.js (hunk 1 above)
Comment (Codebmk) [reply platform/src/common/components/Map/components/Sidebar.js:235]: Yes it's going to be used later

File: platform/src/common/components/Map/components/Sidebar.js (hunk 2 above)
Comment (Codebmk) [reply platform/src/common/components/Map/components/Sidebar.js:755]: It's part of the text. The location name followed by 's

File: platform/src/common/components/Map/components/MapNodes.js (hunk 3 above)
Comment (Codebmk) [reply platform/src/common/components/Map/components/MapNodes.js:111]: Noted

File: platform/src/common/components/Map/components/MapNodes.js [hunk 4]
```diff
@@ -103,6 +115,16 @@ export const getAQIcon = (pollutant, value) => {
...
+
+  return messages[aqiCategory?.icon] || '';
```
Comment (Baalmart) [inline platform/src/common/components/Map/components/MapNodes.js:125]: Hi @Codebmk , what is your view of the optional chaining operator (`?.`)?

Appears that it is supported in modern JavaScript environments, but it's not available in older versions of JavaScript so we might want to rethink things as we go about its usage --- it is supported in Node.js versions 14 and above, and in most modern browsers. However, it's not available in older JavaScript environments.

Comment (Codebmk) [reply platform/src/common/components/Map/components/MapNodes.js:125]: We are using node 14, so this shouldn't be an issue

Comment (Baalmart) [reply platform/src/common/components/Map/components/MapNodes.js:125]: Cool, thanks @Codebmk

**Review by Baalmart — APPROVED** [review]

**Pull request merged**